components/apps.js

```diff
@@ -649,33 +647,152 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
  * Check if this account owns a depot. Only works if enablePicsCache option is enabled and appOwnershipCached event
  * has been emitted.
  * @param {int} depotid
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
  * @returns {boolean}
  */
-SteamUser.prototype.ownsDepot = function(depotid, excludeSharedLicenses) {
-	return this.getOwnedDepots(excludeSharedLicenses).indexOf(parseInt(depotid, 10)) != -1;
+SteamUser.prototype.ownsDepot = function(depotid, filter) {
+	return this.getOwnedDepots(filter).indexOf(parseInt(depotid, 10)) != -1;
 };
 
 /**
- * Returns an array of package IDs this account owns. Only works if enablePicsCache option is enabled and appOwnershipCached event
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of licenses this account owns.
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
-	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
+SteamUser.prototype.getOwnedLicenses = function() {
+	if (!this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	return this.licenses;
+};
+
+/**
+ * Returns an array of package IDs this account owns (different from owned licenses). The filter only
+ * works, if enablePicsCache option is enabled and appOwnershipCached event has been emitted.
+ * @param {object|function} filter - Options for what counts for ownership, or a custom filter function
+ * @returns {int[]}
+ */
+SteamUser.prototype.getOwnedPackages = function(filter) {
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	let packages = this.getOwnedLicenses();
+
+	// If passed filter argument is empty, and global ownership filter is set, use it
+	if (!filter && this.options.ownershipFilter) {
+		filter = this.options.ownershipFilter;
 	}
+
+	// Support for the old syntax and behavior (<= v4.19.12)
+	if (!filter) {
+		// Slight deviation of old behavior: don't return expired licenses
+		return this._returnPackages(packages);
+	}
+	// If filter options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof filter === 'boolean') {
+		let excludeSharedLicenses = filter;
+		filter = {excludeShared: excludeSharedLicenses};
```

Comment:
```suggestion
		filter = {excludeShared: filter};
```

Reply:
I just did that to show `excludeSharedLicenses` is meant here,

Reply:
But you already have a comment here
> If filter options is a boolean, we asssume it's excludeSharedLicenses

(triple s, tho)

Reply:
True, I think either is fine, so I'll just commit it.